### PR TITLE
allow map keys to be templatized

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/9321

### CODE REVIEW

File: pkg/util/stringreplace/object.go [hunk 1]
```diff
@@ -43,9 +43,12 @@ func visitValue(v reflect.Value, visitor func(string) string) {
...
 	case reflect.Map:
 		vt := v.Type().Elem()
-		for _, k := range v.MapKeys() {
-			val := visitUnsettableValues(vt, v.MapIndex(k), visitor)
-			v.SetMapIndex(k, val)
+		for _, oldKey := range v.MapKeys() {
+			newKey := visitUnsettableValues(oldKey.Type(), oldKey, visitor)
+			oldValue := v.MapIndex(oldKey)
+			newValue := visitUnsettableValues(vt, oldValue, visitor)
+			v.SetMapIndex(oldKey, reflect.Value{})
```
Comment (bparees) [inline pkg/util/stringreplace/object.go]: is this really how you "remove" a value from a map?

Comment (deads2k) [inline pkg/util/stringreplace/object.go]: Of course, this is go!

**Comment by bparees** [comment]
lgtm [merge]

**Comment by deads2k** [comment]
yum, re[test]

**Pull request merged**